### DISCUSSION TITLE
WearOS device direct connection experiences

### DISCUSSION THREAD

**Discussion by simonpickering** [j-kaltes/Juggluco#71] [General]
I'm curious to see how well other WearOS devices (aside from the Samsung Galaxy Watch 4: https://www.juggluco.nl/JugglucoWearOS/mgdL/index.html) work as I'm thinking of buying a device and thought I'd see what might have the best chance of working.

If anyone uses a WearOS device with the sensor direct connection (or if you don't because it didn't work very well) I, and I'm sure others, would be interested to hear how well it works for your particular device/sw version. Thanks :)

**Comment by shaneharper123** [comment]
I run Juggluco on a **TicWatch Pro 3 GPS** Smart Watch. I use Libre 2 sensors and I connect to them directly from my watch.

I've been using my current set up for about a year and I am happy with how well it works. I'm still running the last version of Juggluco that was available on Google Play.

I always wear the watch on the wrist of the arm with the sensor. I started doing this because often if I'd go out for a walk with the watch on my left wrist and a sensor on my right arm the watch would lose connection with the sensor. I believe always wearing the watch on the same side as the sensor helps a lot. It was odd at first wearing a watch on my right wrist but now it doesn't bother me.

Most nights while I'm sleeping there'll be periods where the watch doesn't receive sensor readings. Perhaps that's due to some battery saving "feature" of the watch. When I'm not sleeping there is rarely a problem with the connection.
      
I get ~48 hours of battery life on a single charge at the moment on my current watch (my second TicWatch Pro 3) that is around nine months old. While I'm awake I usually have the watch configured to never turn off the OLED screen. (I could get more time on a single charge if I turned that off but I like to convenience of just being able to glance at my watch to see the latest sensor reading without first pressing a button or touching the screen.)

The watch has NFC but I believe that can only be used for mobile payments and it's not possible to use that to scan a new sensor. I scan a new sensor with my phone and then transfer information about the new sensor to my watch. Sometimes this can take a few attempts. (A more recent version of Juggluco might work better.) It seems to help to have the watch on its charger. Also it's necessary to have both the watch and phone connected to a Wifi hotspot.

I am also curious how well other WearOS devices work running Juggluco with a direct sensor connection. In any case I think the TicWatch Pro 3 works really well.

**Comment by j-kaltes** [comment]
I have bought a second hands TicWatch Pro for 30 Euro. It does connect with FreeStyle Libre 2 sensors the same or better than Samsung Galaxy Watch 4, but with Libre 3 sensors it is much worse, missing every second value, so that it receives glucose values every 2 minutes instead of every minute.
The connection between the Phone and the Watch was not very well with the Google Play version. For that reason Phone and Watch in newer version of Juggluco can also communicate via Bluetooth (on Watch 4 that was done automatically by the Wearable app or so).
This TicWatch Pro has also the problem that when you don't move your arm enough the watch goes to sleep and losses contact with the sensor. Putting the watch on the charger is the only solution. See: https://www.reddit.com/r/androiddev/comments/126not8/bluetooth_turned_off_automatically/

**Reply by simonpickering** [reply]
I was going to ask whether the phone was able to step in if the watch missed some readings, good to know it does

**Reply by j-kaltes** [reply]
What do you mean with "phone is able to step in"? Either the phone is connected to the sensor or the watch. The phone doesn't step in when the watch misses a reading when the watch is directly connected to the sensor, except when you scan the sensor with the phone.

**Reply by simonpickering** [reply]
In which case I misunderstood, no worries!
